wfsim/core.py

```diff
@@ -291,89 +291,106 @@ def __call__(self, instruction):
+        p_per_channel[:, np.in1d(channels, config['turned_off_pmts'])] = 0
         
-        self._photon_channels = np.array([]).astype(int)
+        _photon_channels = np.array([]).astype(int)
         for ppc, n in zip(p_per_channel, n_photons):
-            self._photon_channels = np.append(self._photon_channels,
+            _photon_channels = np.append(_photon_channels,
                     np.random.choice(
                         channels,
                         size=n,
                         p=ppc / np.sum(ppc),
                         replace=True))
+        return _photon_channels
 
-    def photon_timings(self, t, n_photons, recoil_type):
+    @staticmethod
+    def photon_timings(t, n_photons, recoil_type, config):
         if n_photons == 0:
             return
 
-        if (self.config.get('s1_model_type') == 'simple' and 
-           recoil_type in NestId.ALL):
+        if (config.get('s1_model_type') == 'simple' and 
+           recoil_type in NestId._ALL):
             # Simple S1 model enabled: use it for ER and NR.
-            self._photon_timings = np.append(self._photon_timings,
-                t + np.random.exponential(self.config['s1_decay_time'], n_photons))
-            self._photon_timings += np.random.normal(0,self.config['s1_decay_spread'],len(self._photon_timings))
-            return
+            _photon_timings = t + np.random.exponential(config['s1_decay_time'], n_photons)
+            _photon_timings += np.random.normal(0,config['s1_decay_spread'],len(_photon_timings))
+            return _photon_timings
 
-        #TODO Fix the other recoil types to refer to the numbers
+        #TODO Looks nice but don't know if this actually works
         try:
-            self._photon_timings = np.append(self._photon_timings,
-                t + getattr(self, recoil_type.lower())(n_photons))
+            recoil_types = [k for k in vars(NestId) if not k.startswith('_')]
+            for recoil in recoil_types:
+                if recoil in NestId.recoil:
+                    _photon_timings = t + getattr(S1, recoil_type.lower())(n_photons=n_photons,
+                                                                           config=config,
+                                                                           singlet_triplet_delays=Pulse.singlet_triplet_delays,
+                                                                           phase=self.phase)
+
         except AttributeError:
             raise AttributeError('Recoil type must be ER, NR, alpha or LED, not %s' % recoil_type)
 
-    def alpha(self, size):
+    @staticmethod
+    def alpha(size,config,singlet_triplet_delays, phase):
         # Neglible recombination time
-        return self.singlet_triplet_delays(size, self.config['s1_ER_alpha_singlet_fraction'])
+        return singlet_triplet_delays(size, config['s1_ER_alpha_singlet_fraction'],config, phase)
 
-    def led(self, size):
+    @staticmethod
+    def led(size,config, **kwargs):
```

Comment:
Why do we need the `kwargs`?

Reply:
Because it allows you to call it in line 350